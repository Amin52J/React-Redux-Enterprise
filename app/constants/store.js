import {routerReducer} from 'react-router-redux/reducer';
import routerMiddleware from 'react-router-redux/middleware';
import createStore from 'redux/lib/createStore';
import applyMiddleware from 'redux/lib/applyMiddleware';
import {createEpicMiddleware} from 'redux-observable/lib/cjs/createEpicMiddleware';
import persistCombineReducers from 'redux-persist/lib/persistCombineReducers';
import storage from 'redux-persist/lib/storage';
import createHistory from 'history/createBrowserHistory';

// rxjs observables
import {ajax} from 'rxjs/observable/dom/ajax';
import {of} from 'rxjs/observable/of';
// ----------------

import hocReducer from '@hoc/cleanOnUnmount/reducer';
import * as reducers from '@reducers';
import epics from '@epics';

// global definitions
import 'rxjs/add/operator/map';
import 'rxjs/add/operator/mergeMap';
import 'rxjs/add/operator/catch';
import 'rxjs/add/operator/takeUntil';
// ------------------

export const history = createHistory();

// the persist config
const persistConfig = {
  key: 'root',
  storage
};

// the epics middleware
const epicMiddleware = createEpicMiddleware(epics, {
  dependencies: {
    getJSON: ajax.getJSON,
    of
  }
});

// the reducers
const combinedReducers = persistCombineReducers(persistConfig, {
  ...reducers,
  routing: routerReducer
});
const reducer = hocReducer(combinedReducers); // adds reset reducer to our apps reducer
const store = createStore(reducer, applyMiddleware(epicMiddleware, routerMiddleware(history)));

export default store;
