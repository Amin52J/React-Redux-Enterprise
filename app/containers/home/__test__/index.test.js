import React from 'react';
import { mount } from 'enzyme';
import Home from '../index';
import store, { history } from '../../../constants/store';
import { ConnectedRouter } from 'react-router-redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

describe('Home Container - Container', () => {
  it('renders with no problem', () => {
    const component = mount(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <Home />
        </ConnectedRouter>
      </Provider>,
    );
    const tree = renderer.create(component).toJSON();

    expect(
      component
        .find('div')
        .first()
        .hasClass('home-page'),
    ).toEqual(true);
    expect(tree).toMatchSnapshot();
  });
});
