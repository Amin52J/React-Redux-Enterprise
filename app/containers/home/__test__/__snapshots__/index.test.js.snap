// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Container - Container renders with no problem 1`] = `
<div
  className="home-page sc-htpNat bwjOlV"
>
  Home Page
  <div
    className="test-component sc-bwzfXH jPKXJw"
  >
    Test Component
    <div
      className="test-element sc-bdVaJa hXWJrw"
    >
      Test Element
    </div>
  </div>
</div>
`;
